app/views/apps/show.html.erb

```diff
@@ -0,0 +1,49 @@
+<style>
+  #Qualifications {
+    width: 300px;
+    height: 100px;
+  }
+</style>
+
+<h1><%=@app.name%>'s Application</h1>
+<p><u>Address</u>: <%=@app.address%>, <%=@app.city%>, <%=@app.state%>, <%=@app.zip%></p>
+<p><u>Description</u>: <%=@app.description%></p>
+<p><u>Status</u>: <%=@app.status%></p>
+
+
+<% @app.pets.each do |pet| %>
+  <p><u>Pets on your application</u>: <%= link_to "#{pet.name}", "/pets/#{pet.id}" %></p>
+<% end %>
+
+
+<% if @app.status == "In Progress" %>
+<h2>Search a pet</h2>
+  <%= form_with url: "/apps/#{@app.id}?pet_name=#{params[:pet_name]}&commit=Search", method: :get do %>
+  <%= text_field_tag "pet_name", params[:pet_name], placeholder: "Search pets by name here" %>
+  <%= submit_tag "Submit" %>
+  <% end %>
+<% end %>
+
+  <% if params[:pet_name].present? %>
+    <h3>Pets matching search</h3>
+    <% @pets = Pet.search(params[:pet_name]) %>
+    <% @pets.each do |pet| %>
+      <p><%= link_to "#{pet.name}", "/pets/#{pet.id}" %></p>
+      <% if @app.pets.include?(pet) %>
+        <p>This pet is already on this application</p>
+      <% else %>
+        <%= button_to "Adopt #{pet.name}", "/apps/#{@app.id}?app_id=#{params[:id]}&pet_id=#{pet.id}&commit=Add+Pet", method: :patch %>
+      <% end %>
+    <% end %>
+  <% end %>
+
+<% if @app.pets.count > 0 && @app.status == "In Progress"%>
+<h3>Submit my application</h3>
+  <%= form_with url: "/apps/#{@app.id}?commit=Submit+Application", method: :patch, local: true do |form|%>
```

Comment:
By making the above change, this also means you don't need this update action to distinguish between submitting application and adding a pet!